### PR TITLE
Removed EY2G function, added TY2G function, fixed TY2Cv function, add…

### PR DESCRIPTION
**Pull request by ldowen**
…ed missing GammaLaw functions

### CODE REVIEW

**Review by drummerdoc — APPROVED** [review]
Looks ok to me

**Comment by ldowen** [comment]
I just realized we also need a `RYP2T` function. Should I add that too before completing this PR?

**Comment by drummerdoc** [comment]
Yeah, seems so.

**Review by drummerdoc — APPROVED** [review]
looks good

**Comment by ldowen** [comment]
Sorry to keep noting small changes. I am going through PeleC to see what kind of impact these changes will have and noticing other functions that I didn't see before. The `RYET2P` function might have a few issues. Seems like it is actually trying to be `RYE2TP` and energy is passed by reference but should not be modified.

**Comment by drummerdoc** [comment]
Shouldn't R and Y also be passed by reference into RYE2TP and not be modified?

**Comment by ldowen** [comment]
Typically, we pass variables that aren't modified by value and we pass variables that are modified by reference. So the name should be changed to `RYE2TP` and E should be passed by value. Or the function should be eliminated altogether unless it will be needed for the real gas EOS functions.

**Comment by drummerdoc** [comment]
Oh, thanks.  didn't have time to check.  That suggests maybe that @jrood-nrel is using this differently than you think?  Jon, can you comment?

**Comment by jrood-nrel** [comment]
Aren't all variables expected to be pass by reference since it was in Fortran previously?

**Comment by drummerdoc** [comment]
1. I'd prefer everything by reference only because it looks nicer.  But that would probably "break" a lot of code.  We can do that later, maybe?
2. The real question there is how you are using that function `RYET2P` in PeleC.  If Landon makes E read-only will that break your code?

**Comment by ldowen** [comment]
Typically, we would pass everything by reference and just label things that don't change as `const`. However, I assumed there is some issue with doing that in the Fuego routines.

**Comment by drummerdoc** [comment]
There is, as long as we still support F90.

File: Eos/GammaLaw/EOS.H [hunk 1]
```diff
@@ -151,6 +161,15 @@ Y2X(amrex::Real Y[], amrex::Real X[])
...
 
+AMREX_GPU_HOST_DEVICE
+AMREX_FORCE_INLINE
+void
+X2Y(amrex::Real X[], amrex::Real Y[])
+{
+  for (int n = 0; n < NUM_SPECIES; n++)
+    Y[n] = X[n];
+}
+
```
Comment (EnnaDelfen) [inline Eos/GammaLaw/EOS.H:182]: Hum... Excuse me ? What is that ?

**Comment by jrood-nrel** [comment]
I don't see E getting modified in `RYET2P` so it seems fine to me. I don't know why we are passing it by reference at the moment. Maybe because Fuego expects it? It uses so many pointers. I'm assuming it was an oversight when we moved from Fortran.

**Comment by drummerdoc** [comment]
I think there will some cleanup through all this again once we no longer have a need for F90 code anywhere.  There is still a bunch in PelePhysics and PeleLM for now though.

**Review by EnnaDelfen — CHANGES_REQUESTED** [review]
explain why X=Y please ! Thanks

File: Eos/GammaLaw/EOS.H (hunk 1 above)
Comment (drummerdoc) [reply Eos/GammaLaw/EOS.H:182]: Gamma law gases don't have molecular weights, so the conversion is nonsensical.  This just ensures that if Sum(X)=1, so does Sum(Y), and preserves all the things that should be preserved.

Comment (EnnaDelfen) [reply Eos/GammaLaw/EOS.H:182]: well there still is a law to convert, X/Xk ends up being 1 but it's still supposed to be valid... But I understand, I did not see this was for the GammaLaw so I guess we're good

**Comment by EnnaDelfen** [comment]
Should we merge ? Everybody good with this ?

**Comment by drummerdoc** [comment]
I'm good.  Merge please

**Pull request merged**